### DISCUSSION TITLE
when using a url: entry in the TOC: how to open it in a separate tab/window

### DISCUSSION THREAD

**Discussion by parmentelat** [executablebooks/meta#1173] [Q&A] [answered]
I just came across the ability to add entries of the form url: to other URLs right in the TOC
in https://jupyterbook.org/en/stable/structure/toc.html#types-of-content-entries

this is very useful !
I'd like such pages to open in a separate tab/window though (according to the user's preferences)
because right now the page opens in the current browser and we lose context
in other words users need to "go back", and I fear I will lose a portion of my class a that point

is there currently a way to tweak that ? I was thinking maybe something like
```
- url: https://some.other.site/
  title: an external topic
  target: _
```
inspired from the `<a>` tag, or any other flag to set like `spawn: true` or whatever else

I could not find that in the doc (not sure if the above mentioned section is the only place where this is mentioned, or if I should look for a more 'reference-oriented' section some place else...)

if that's not yet possible, how does that sound ?

**Comment by agoose77** [comment, marked as answer]
The answer *right now* is that you can use custom JS to set the target on these pages:
```javascript
document.querySelectorAll("nav a.reference.external").forEach(node => {node.target = "_blank"})
```

Then you can load this JS on each page: https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/static_assets.html#define-a-list-of-assets-in-conf-py

For JupyterBook, you would set this in your `_config.yml` e.g. https://jupyterbook.org/en/stable/advanced/html.html#use-plausible-analytics.

For MyST-MD, we don't yet support external URLs in the main table of contents, though it is something that we're thinking about: executablebooks/mystmd#1188

**Reply by parmentelat** [reply]
thanks a million for spelling out the steps to pour my own javascript into the mix, it does the trick for me at this point :)
(plus: I've already opened the pandora box and started to add other custom stuff in there...;)

now maybe on the longer run it might help to be able to customize this without having to resort to js
maybe even at a global level rather than my initial suggestion because, on second thought, this setting would conveniently apply to the whole document
but that's for another time I take it
thanks again !
